rtcpeerconnection.go

```diff
@@ -23,47 +25,85 @@ import (
 // comparisons when no value was defined.
 const Unknown = iota
 
-// RTCPeerConnection represents a WebRTC connection between itself and a remote peer
+// RTCPeerConnection represents a WebRTC connection that establishes a
+// peer-to-peer communications with another RTCPeerConnection instance in a
+// browser, or to another endpoint implementing the required protocols.
 type RTCPeerConnection struct {
 	sync.RWMutex
 
 	configuration RTCConfiguration
 
-	// ICE
-	OnICEConnectionStateChange func(ice.ConnectionState)
-	IceConnectionState         ice.ConnectionState
-
-	networkManager *network.Manager
-
-	// Signaling
+	// CurrentLocalDescription represents the local description that was
+	// successfully negotiated the last time the RTCPeerConnection transitioned
+	// into the stable state plus any local candidates that have been generated
+	// by the IceAgent since the offer or answer was created.
 	CurrentLocalDescription *RTCSessionDescription
+
+	// PendingLocalDescription represents a local description that is in the
+	// process of being negotiated plus any local candidates that have been
+	// generated by the IceAgent since the offer or answer was created. If the
+	// RTCPeerConnection is in the stable state, the value is null.
 	PendingLocalDescription *RTCSessionDescription
 
+	// CurrentRemoteDescription represents the last remote description that was
+	// successfully negotiated the last time the RTCPeerConnection transitioned
+	// into the stable state plus any remote candidates that have been supplied
+	// via AddIceCandidate() since the offer or answer was created.
 	CurrentRemoteDescription *RTCSessionDescription
+
+	// PendingRemoteDescription represents a remote description that is in the
+	// process of being negotiated, complete with any remote candidates that
+	// have been supplied via AddIceCandidate() since the offer or answer was
+	// created. If the RTCPeerConnection is in the stable state, the value is
+	// null.
 	PendingRemoteDescription *RTCSessionDescription
 
-	idpLoginURL *string
+	// SignalingState attribute returns the signaling state of the
+	// RTCPeerConnection instance.
+	SignalingState RTCSignalingState
+
+	// IceGatheringState attribute returns the ICE gathering state of the
+	// RTCPeerConnection instance.
+	// IceGatheringState RTCIceGatheringState // FIXME NOT-USED
+
+	// IceConnectionState attribute returns the ICE connection state of the
+	// RTCPeerConnection instance.
+	// IceConnectionState RTCIceConnectionState  // FIXME SWAP-FOR-THIS
+	IceConnectionState ice.ConnectionState // FIXME REMOVE
```

Comment:
What is wrong with having ice.ConnectionState be public?

We have to maintain mappings and boilerplate otherwise. I think conceptually we want `github.com/pions/webrtc` to be tiny, it is just the shim overtop while everything complicated lives inside pkg/internal

Reply:
Ohhh.... There is nothing wrong with this stuff. The actual spec keeps it separate. The `RTCIceConnectionState` is part of the WebRTC interface and the `IceAgent` has its own state management. I started working towards it but needed to fix up sctp a little to be able to finish `RTCIceTransport` which is the abstraction layer for the `IceAgent`. They literally do it that way in order to not have what we ended up doing which is sharing that enum in two separate packages. The webrtc and the ice. So I am just following the spec but cannot swap it right away in order to keep the PRs still functional and not break the library. Sorry.